### DISCUSSION TITLE
Ad-Hoc document approval scenario support?

### DISCUSSION THREAD

**Discussion by StefanCatalinP** [elsa-workflows/elsa-core#3397] [Q&A]
Does Elsa 2 have any support for an 'Ad-Hoc' document approval scenario? 

What I mean by 'Ad-Hoc' would be something like:
An instance of a workflow definition is in progress and waiting for a user action on a conditional flow step, the user decides on the spot that for this particular instance the workflow definition needs to be extended with a new branch on the conditional flow step.
Can a workflow instance's definition be changed mid-execution? And then have that instance resume from the previous step before the update on the definition, and of course continue execution on the new definition version.

**Comment by work-flower** [comment]
This was possible in WF with mapping older workflow definition to the new version. Curious about suggestions on this.

**Comment by FransVanEk** [comment]
Hey @StefanCatalinP and @work-flower,

Currently, I am looking into the options of doing just that. Migrating a workflow instance to a new version. From what I have investigated until now the preliminary conclusions are

- the engine cannot decide what to do. it can assume things but needs to be told how the migration should look like. 
- It can be tricky due to multiple active activities in the workflow instance. Like after a fork.
- signals, timers, etc need to be temporarily ignored during the migration. Which can lead to unexpected behavior in the systems that send the signals

So it has some challenges but it is being investigated.

**Reply by work-flower** [reply]
Just as a reference from WF

https://learn.microsoft.com/en-us/dotnet/framework/windows-workflow-foundation/dynamic-update

If Elsa uses the same code base then it should be moderate to include in. I've been in similar cases, please let me know if I can do anything.
